apps/ensindexer/src/config/types.ts

```diff
@@ -0,0 +1,236 @@
+   * - The URL must be a valid URL (localhost urls are allowed)
+   */
+  ensAdminUrl: string;
+
+  /*
+   * This is a namespace for the tables that the indexer will create to store indexed data.
+   * It should be a string that is unique to the running indexer instance.
+   *
+   * Keeping the database schema unique to the indexer instance is important to
+   * 1) speed up indexing after a restart
+   * 2) prevent data corruption from multiple indexer app instances writing state
+   *    concurrently to the same db schema
+   *
+   * No two indexer instances can use the same database schema at the same time.
+   *
+   * Read more about database schema rules here:
+   * https://ponder.sh/docs/api-reference/database#database-schema-rules
+   *
+   * Invariant: Must be a valid non-empty string
+   */
+  ponderDatabaseSchema: string;
+
+  /**
+   * Identify which indexer plugins to activate (see `src/plugins` for available plugins)
+   * This is a set of one or more available plugin names.
+   *
+   * Invariants:
+   * - A set of valid plugin names with at least one value
+   * - For any requested plugin, the config will have a key for chainId in the
+   *   indexedChains object. In that indexedChains object, the config will have a
+   *   rpcEndpointUrl that is defined for that chainId and a rpcMaxRequestsPerSecond
+   *   that is defined for that chainId. This is ensured by the validateChainConfigs
+   *   function in `validations.ts` and not part of the schema validation here.
+   */
+  plugins: Set<PluginName>;
+
+  /**
+   * A feature flag to enable or disable healing of addr.reverse subnames.
+   * If this is set to true, ENSIndexer will attempt to heal subnames of
+   * addr.reverse.
+   *
+   * If this is not set, the default value is set to `DEFAULT_HEAL_REVERSE_ADDRESSES`.
+   *
+   * Setting this to `true` results in indexed data no longer being backwards
+   * compatible with the ENS Subgraph. For full data-level backwards
+   * compatibility with the ENS Subgraph, this should be set to `false`.
+   *
+   * Invariant: The value must be 'true' or 'false'
+   */
+  healReverseAddresses: boolean;
+
+  /**
+   * The network port ENSIndexer listens for http requests on. ENSIndexer
+   * exposes various APIs, most notably the GRAPHQL API. All APIs are accessible
+   * on the same port. This defaults to DEFAULT_PORT if not specified.
+   *
+   * Invariants:
+   * - The port must be an integer between 1 and 65535
+   */
+  port: number;
+
+  /**
+   * The endpoint URL for the ENSRainbow API. ENSIndexer uses this for fetching
+   * data about labelhashes (healing) when indexing. This should be set to a
+   * colocated instance of ENSRainbow for best performance.
+   *
+   * Invariant: The URL must be a valid URL. localhost urls are allowed,
+   * and expected.
+   */
+  ensRainbowEndpointUrl: string;
+
+  /**
+   * Configuration for each indexed chain, keyed by chain ID.
+   *
+   * Invariants:
+   * - Each key (chainId) must be a number
```

Comment:
We should make this more strict. For example:

```suggestion
   * - Each key (chainId) must be a non-negative integer
```

I'm not sure if it must be positive or non-negative?